src/scripts/modules/countdown-timer.js

```diff
@@ -0,0 +1,111 @@
+        return;
+      }
+
+      const calculate = () => {
+        let startDate = new Date();
+        startDate = startDate.getTime();
+
+        // turn milliseconds into seconds to make calculations easier
+        let timeRemaining = parseInt((endDate - startDate) / 1000, 10);
+
+        if (timeRemaining < 0) {
+          finished = true;
+          return;
+        }
+
+        // divide time remaining by number of seconds in a day & remove days from timeRemaining
+        days = parseInt(timeRemaining / 86400, 10);
+        timeRemaining %= 86400;
+
+        // divide time remaining by number of seconds in an hour & remove hours from timeRemaining
+        hours = parseInt(timeRemaining / 3600, 10);
+        timeRemaining %= 3600;
+
+        // divide time remaining by number of seconds in a min & remove mins from timeRemaining.
+        minutes = parseInt(timeRemaining / 60, 10);
+        timeRemaining %= 60;
+
+        seconds = parseInt(timeRemaining, 10);
+
+        daysElement.innerHTML = parseInt(days, 10);
+        hoursElement.innerHTML = (`0${hours}`).slice(-2);
+        minutesElement.innerHTML = (`0${minutes}`).slice(-2);
+        secondsElement.innerHTML = (`0${seconds}`).slice(-2);
+      };
+
+      const timerFinished = () => {
+        // HTTP request for retreiving reveal information
+        const xhr = new XMLHttpRequest();
+        xhr.onreadystatechange = function revealContent() {
+          if (this.readyState === 4) {
+            const jsonResponse = JSON.parse(this.responseText);
+
+            const wrapper = document.createElement('div');
+            wrapper.classList.add('coutdownTimer-revealContainer');
+
+            if (jsonResponse.attrs.revealTitle !== '') {
+              const title = document.createElement('h2');
+              title.innerText = jsonResponse.attrs.revealTitle;
+              title.classList.add('countdownTimer-title');
+              wrapper.append(title);
+            }
+
+            if (jsonResponse.attrs.revealContent !== '') {
+              const content = document.createElement('p');
+              content.innerText = jsonResponse.attrs.revealContent;
+              content.classList.add('countdownTimer-content');
+              wrapper.append(content);
+            }
+
+            if (jsonResponse.attrs.revealBtnUrl !== '' && jsonResponse.attrs.revealBtnText !== '') {
+              const button = document.createElement('a');
+              button.href = jsonResponse.attrs.revealBtnUrl;
+              button.innerText = jsonResponse.attrs.revealBtnText;
+              button.classList.add('btn', 'countdownTimer-btn');
+              wrapper.append(button);
+            }
+
+            timer.classList.add('is-finished');
+            document.querySelector(`.countdownTimer[data-ref="${timerRef}"]`).append(wrapper);
+          }
+        };
+
+        xhr.open('GET', `/wp-json/benenson/v1/reveal-content/${benensonData.post_id}/${timerRef}`);
```

Comment:
We can't rely on the REST API base being `/wp-json/`. If the site doesn't have pretty permalinks enabled, this will fail. I'd suggest localising the script with `'rest_base' => get_rest_url(),`, and use that as a the foundation for building up this route.